tests/test_calculating_n50.py

```diff
@@ -0,0 +1,296 @@
+                                1 has blanks
+                                1 has random text paragraph
+
+4. test_assem4.fasta - 3 invalid contigs:
+                            1 has no sequence
+                            1 has blanks
+                            1 has random text paragraph
+
+5. test_assem5.fasta - Random text paragraph but will pass
+        as a valid assembly due to the nature of embedded text
+
+6. test_assem6.fasta - Random text paragraph which won't pass
+
+7. test_assem7.fasta - Orphaned nucleotides (valid sequences
+        without fasta headers). This won't pass
+
+8. test_assem8.fasta - Empty file (with a valid fasta extension)
+'''
+
+# Defining return objects for successful test functions'''
+ 
+'''Dictionary of filenames with value of 0/1 indicating unzipped/zipped'''
+test_file_dic_good = {
+    'test_assem1.fasta': 0, 'test_assem2.fasta': 0, 'test_assem3.fasta': 0,
+    'test_assem4.fasta': 0, 'test_assem5.fasta': 0, 'test_assem6.fasta': 0,
+    'test_assem7.fasta': 0, 'test_assem8.fasta': 0
+}
+
+'''List of dictionaries. Each dictionary corresponds to one assembly file
+with contig names (keys) and lengths (values)'''
+test_assem_list_good = [
+    {'contig1': 8, 'contig2': 5, 'contig3': 4, 'contig4': 1, 'contig5': 7,
+     'contig6': 2, 'contig7': 3, 'contig8': 10, 'contig9': 6, 'contig10': 9},
+    {'contig1': 1, 'contig2': 2, 'contig3': 3, 'contig4': 4, 'contig5': 5},
+    {'contig1': 5, 'contig2': 150, 'contig3': 12},
+    {'I': 1}
+]
+
+'''Tuple of N50 values and their corresponding logarithms with base 10.
+Each tuple element is a list of N50 values for all valid assembly files'''
+test_n50_tuple_good = (
+    [7, 4, 150, 1],
+    [0.845, 0.602, 2.176, 0.0]
+)
+
+'''List of summary statistics of N50 values for all assembly files'''
+test_n50_stat_summary_good = [4, 1, 150, 40.5, 5.5, 73.041]
+
+'''Defining pre-existing histogram plot of log N50 values'''
+expected_fig = os.path.join(TEST_INPUT_DIR, 'expected.png')
+
+'''Creating good and bad combinations of organism names'''
+genus_good = ['Homo', 'H.', 'H']
+species_good = ['sapiens', 'sap']
+genus_bad = ['homo', 'H,', 'Homo[Organism]']
+species_bad = ['sap[Org]', 'sap.', 'Sapiens']
+
+
+class TestCalcN50:
+    @classmethod
+    def setup_class(cls):
+        sys.path.append(os.path.dirname(os.path.abspath(__file__)))
+
+
+    # Test for checking good fasta file name extensions
+    def test_fasta_extn_check_true(self):
+        test_file_dic = {}
+        
+        '''Creating dictionary of 8 fasta file names with value 0'''
+        for i in range(1, 9):
+            filename = 'test_assem' + str(i) + '.fasta'
+            test_file_dic[filename] = 0
+        
+        method_dic = fasta_extn_check(TEST_INPUT_DIR)
+        assert test_file_dic == method_dic
+
+
+    # Test for checking fasta files in empty directory
+    def test_fasta_extn_check_empty(self):
+        test_file_dic = {}
+        method_dic = fasta_extn_check(EMPTY_DIR)
```

Comment:
This test fails for with a `FileNotFoundError` as the `EMPTY_DIR` does not exist. To Fix this you can create an empty `.gitkeep` file in that directory and then do a `git add EMPTY_DIR` where `EMPTY_DIR` is the path.